ServerCore/Pages/Puzzles/PlayPuzzles.cshtml

```diff
@@ -0,0 +1,368 @@
+﻿@page "/{eventId}/{eventRole}/PlayPuzzles"
+@model ServerCore.Pages.Puzzles.PlayPuzzlesModel
+
+@{
+    @using Helpers;
+
+    ViewData["Title"] = "PlayPuzzles";
+    ViewData["AdminRoute"] = "/Puzzles/Index";
+    ViewData["AuthorRoute"] = "/Puzzles/Index";
+
+    bool unsolvedFilter = Model.StateFilter == PlayPuzzlesModel.PuzzleStateFilter.Unsolved;
+    bool canSubmit = DateTime.UtcNow >= Model.Event.EventBegin;
+}
+<style>
+    .radioButton {
+        padding: 0px 5px;
+        color: inherit;
+        text-decoration: none;
+    }
+
+        .radioButton:hover {
+            color: inherit;
+            text-decoration: none;
+            cursor: pointer;
+        }
+</style>
+
+@if (DateTime.UtcNow > @Model.Event.AnswerSubmissionEnd)
+{
+    <div class="alert alert-warning" role="alert">
+        <h4>The event is over.</h4>
+        <span>You may check answers, but they will not count toward your ranking.</span>
+    </div>
+}
+
+<h2>Team puzzles</h2>
+@if (Model.Team == null)
+{
+    <p>You cannot see team puzzles because you are not yet on a team!</p>
+}
+else if (!Model.VisibleTeamPuzzleViews.Any())
+{
+    <p>Looks like there are no Team puzzles unlocked yet.</p>
+}
+else
+{
+    <div>
+        <a asp-page="/Teams/Answers" asp-route-teamId="@Model.Team.ID">View All Correct Team Answers</a>
+    </div>
+    <br />
+    <div>
+        <a asp-page="./PlayPuzzles"
```

Comment:
Maybe just one of these all/unsolved links, above both lists?

Reply:
Fixed in newest iteration